packages/react-components/react-rating-preview/src/components/RatingItem/useRatingItem.tsx

```diff
@@ -33,6 +33,7 @@ export const useRatingItem_unstable = (props: RatingItemProps, ref: React.Ref<HT
   const root = slot.always(
     getIntrinsicElementProps('span', {
       ref: useMergedRefs(useFocusWithin<HTMLSpanElement>(), ref),
+      'aria-label': `${value} star${value === 1 ? '' : 's'}`,
```

Comment:
The labels need to go on the inputs, not the root.

```suggestion
```

For the fullValueInput:
```
      'aria-label': context.itemLabelFormat ? context.itemLabelFormat(value) : `${value}`,
```

For the halfValueInput:
```
      'aria-label': context.itemLabelFormat ? context.itemLabelFormat(value - 0.5) : `${value - 0.5}`,
```

Reply:
Or make it a little cleaner by setting a default for `itemLabelFormat` when destructuring from context:

```ts
const { itemLabelFormat = (value => `${value}`) } = context;
```

```ts
    'aria-label': itemLabelFormat(value),
```